testing/trino-tests/src/test/resources/sql/presto/tpcds/hive/unpartitioned/q13.plan.txt

```diff
@@ -3,23 +3,24 @@ final aggregation over ()
         remote exchange (GATHER, SINGLE, [])
             partial aggregation over ()
                 join (INNER, REPLICATED):
-                    join (INNER, REPLICATED):
-                        join (INNER, REPLICATED):
-                            join (INNER, REPLICATED):
+                    join (INNER, PARTITIONED):
```

Comment:
I'm assuming that this is a side-effect of the expression rewriting in tpcds `q13.sql`

**INPUT NODE in `UnwrapCastInComparison**
```
(CAST("ss_net_profit" AS decimal(12, 2)) BETWEEN CAST(DECIMAL '100.00' AS decimal(12, 2)) AND CAST(DECIMAL '200.00' AS decimal(12, 2)))
```

**OUTPUT NODE BEFORE CHANGES**
```
(CAST("ss_net_profit" AS decimal(12, 2)) BETWEEN CAST(DECIMAL '100.00' AS decimal(12, 2)) AND CAST(DECIMAL '200.00' AS decimal(12, 2)))
```

**OUTPUT NODE AFTER CHANGES**
```
("ss_net_profit" BETWEEN CAST(DECIMAL '100.00' AS decimal(7, 2)) AND CAST(DECIMAL '200.00' AS decimal(7, 2)))
```

Reply:
looks reasonable

to be on the safe side, we will want to confirm this change with benchmarks (cc @przemekak )
but i don't think we're ready yet